### DISCUSSION TITLE
Share my code to EdgeX Foundry: Siemens S7 ISO-on-TCP device service

### DISCUSSION THREAD

**Discussion by jiekechoo** [edgexfoundry/community#184] [Ideas]
I have applied this `S7 device service ` in my project, it works.

https://github.com/yiqisoft/device-s7-go

I don't know how to sumbit my code to a new reposisitory. Anyone can help me ?

**Comment by cloudxxx8** [comment]
Are you or your team able to maintain this repository in the future?  We can open a repository in the holding area for you.
https://github.com/edgexfoundry-holding
The first thing is to upgrade the code to v3 by using device-sdk-go main branch

**Reply by jiekechoo** [reply]
OK. got it. I'll upgrade to v3. And our team is able to maintain the `S7` respository.

**Reply by cloudxxx8** [reply]
What is your timezone?  Are you able to join the TSC or Device Core WG meeting?

**Reply by jiekechoo** [reply]
My timezone is Asia/Shanghai. I prefer to join the TSC, but my English is not so good.

**Reply by cloudxxx8** [reply]
It's not required to join the TSC meeting, because it's quite late.  Please join the next Device WG meeting, and we can discuss in Chinese.  It's Oct-17 8:00 AM China time.
https://zoom.us/j/314411765?pwd=TE1PNmNiLzR6cVR6UDJLRkdtRWNtZz09

Please help answer a few questions fow now:
1) Could you provide documentation for the service?  This is an example: https://docs.edgexfoundry.org/3.1/microservices/device/services/device-virtual/Ch-VirtualDevice/
2) what is the open source license this will be contributed under?  Apache 2 is preferred.
3) what physical devices have this been tested against?  What type of simulation is provided to allow adopters to try to use this service?  Could you please document the use of simulation?

Thanks.

**Reply by jiekechoo** [reply]
S7 Device Service for EdgeX V3.x first commit: https://github.com/yiqisoft/device-s7-go/tree/v3, it works.
![image](https://github.com/edgexfoundry/community/assets/3350211/042e8c22-59db-4501-8466-bd7ca292d27f)

for EdgeX V2.x in branch v2: https://github.com/yiqisoft/device-s7-go/tree/v2

**Comment by jiekechoo** [comment]
> It's not required to join the TSC meeting, because it's quite late. Please join the next Device WG meeting, and we can discuss in Chinese. It's Oct-17 8:00 AM China time.
https://zoom.us/j/314411765?pwd=TE1PNmNiLzR6cVR6UDJLRkdtRWNtZz09

OK, I'll join the meeting.

> Could you provide documentation for the service? This is an example: > https://docs.edgexfoundry.org/3.1/microservices/device/services/device-virtual/Ch-VirtualDevice/

Yes, I can provide docs like `VirtualDevice`.

> what is the open source license this will be contributed under? Apache 2 is preferred.

Yes, I prefer Apache V2.

> what physical devices have this been tested against? What type of simulation is provided to allow adopters to try to use this service? Could you please document the use of simulation?

I use S7-1200, S7-300, S7-400 and S7-1500 devices, not a simulator.

**Reply by cloudxxx8** [reply]
thanks, @jiekechoo 
@lindseysimple please help request a repository in the holding area

**Reply by lindseysimple** [reply]
Support ticket created.

**Reply by jiekechoo** [reply]
I didn’t find any repository in holding area.

**Reply by lindseysimple** [reply]
@jiekechoo Since the support ticket was just created this morning and hasn't been processed yet, I'll keep you informed once the repository has been created.
Thanks.

**Reply by lindseysimple** [reply]
Hi @jiekechoo Repository created: https://github.com/edgexfoundry-holding/device-s7

**Reply by jiekechoo** [reply]
I have pushed code to repository for EdgeX v2.x(branch v2) and v3.x(branch main).

**Reply by cloudxxx8** [reply]
please don't push the code directly.
you have to open PR for review
https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines
we won't release v2 anymore, so v3 is enough.

**Comment by jiekechoo** [comment]
Could you change the repository to ‘device-s7-go’, because it was written by golang.

**Reply by cloudxxx8** [reply]
we remove the programming language suffix in the new device services, because the user doesn't need to know the language, such as device-gpio, device-usb-camera.

**Reply by jiekechoo** [reply]
Noted.

**Comment by cloudxxx8** [comment]
close this discussion since the repository has been created
